--- lmpcm_usb-0.5.7/lmpcm_usb.c.orig	2009-05-03 14:02:09.000000000 +0200
+++ lmpcm_usb-0.5.7/lmpcm_usb.c	2009-05-03 14:21:55.000000000 +0200
@@ -30,7 +30,7 @@
 #include <linux/module.h>
 #include <linux/init.h>
 #include <linux/usb.h>
-
+#include <linux/version.h>
 
 #define DRIVER_VERSION	"v0.5.7"
 #define DRIVER_AUTHOR	"David Oliveira <d.oliveira@prozone.org>"
@@ -225,7 +225,11 @@
 
 static int usb_lmpcm_open(struct input_dev *dev) {
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 26)
+	lmpcm_t *mouse = dev->private;
+#else
 	lmpcm_t *mouse = input_get_drvdata(dev);
+#endif
 
 	if (mouse->open++)
 		return 0;
@@ -243,9 +247,11 @@
 
 static void usb_lmpcm_close(struct input_dev *dev) {
 
-	//lmpcm_t *mouse = dev->private;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 26)
+	lmpcm_t *mouse = dev->private;
+#else
 	lmpcm_t *mouse = input_get_drvdata(dev);
-
+#endif
 
 	if (!--mouse->open)
 		usb_kill_urb(mouse->urb);
@@ -273,6 +279,25 @@
 	for ( x = 0 ; keys[x] ; x++ )
 		set_bit(keys[x],inputdev->keybit);
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 24)
+	// Add basic buttons
+
+	inputdev->keybit[LONG(BTN_MOUSE)] = BIT(BTN_LEFT) | BIT(BTN_RIGHT) | BIT(BTN_MIDDLE) |
+					    BIT(BTN_SIDE) | BIT(BTN_EXTRA);
+
+	// Add move mouse movement (X/Y)
+
+	inputdev->relbit[0] = BIT(REL_X) | BIT(REL_Y);
+
+	// Add wheel
+
+	inputdev->relbit[0] |= BIT(REL_WHEEL);
+
+
+	// Private data structure
+
+	inputdev->private = mouse;
+#else
 	// Add basic buttons
 
 	inputdev->keybit[BIT_WORD(BTN_MOUSE)] = BIT_MASK(BTN_LEFT) | BIT_MASK(BTN_RIGHT) | BIT_MASK(BTN_MIDDLE) |
@@ -290,7 +315,7 @@
 	// Private data structure
 
 	input_set_drvdata(inputdev, mouse);
-
+#endif
 	// Input file operations
 
 	inputdev->open = usb_lmpcm_open;
@@ -319,7 +344,7 @@
 
 	struct usb_endpoint_descriptor *endpoint;
 	lmpcm_t *mouse;
-	int pipe, maxp;
+	int pipe, maxp, error;
 	char *buf;
 
 
@@ -388,8 +413,16 @@
 
 	// Register input device
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 26)
 	input_register_device(mouse->inputdev);
-
+#else
+	error = input_register_device(mouse->inputdev);
+	if (error) {
+		input_free_device(mouse->inputdev);
+		lmpcm_free(mouse);
+		return error;
+	}
+#endif
 
 	printk(KERN_INFO "lmpcm_usb: Detected device: %s\n", mouse->name);
 
